cpp/core/convert_core_json_to_marc.cc

```diff
@@ -0,0 +1,386 @@
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <iostream>
+#include <set>
+#include <unordered_map>
+#include "FileUtil.h"
+#include "JSON.h"
+#include "KeyValueDB.h"
+#include "MARC.h"
+#include "MiscUtil.h"
+#include "TextUtil.h"
+#include "TimeUtil.h"
+#include "UBTools.h"
+#include "util.h"
+
+
+namespace {
+
+
+[[noreturn]] void Usage() {
+    ::Usage("[--create-unique-id-db|--ignore-unique-id-dups][--935-entry=entry] --sigil=project_sigil json_input [unmapped_issn_list marc_output]\n"
+            "\t--create-unique-id-db: This flag has to be specified the first time this program will be executed only.\n"
+            "\t--ignore-unique-id-dups: If specified MARC records will be created for unique ID's which we have encountered\n"
+            "\t                         before.  The unique ID database will still be updated.\n"
+            "\t--935-entry: The structure of this repeatable flag is \"(TIT|LOK):subfield_a_value\".  If TIT has been specified then no subfield 2\n"
+            "\t             will be generated.  If LOK has been specified, subfield 2 will be set to LOK.\n"
+            "\t--sigil: This is used to generate an 852 field which is needed by the K10+ to be able to assign records to the appropriate\n"
+            "\t         project.  An example would be DE-2619 for criminology.\n"
+            "\tunmapped_issn_list (output): Here we list the ISSN's for which we have no entry in issns_to_journaltitles_and_ppns.map,\n"
+            "\t                             required unless --extract-and-count-issns-only was specified!\n"
+            "\tmarc_outpt: required unless --extract-and-count-issns-only was specified!\n\n");
+}
+
+
+struct JournalTitlePPNAndOnlineISSN {
+    std::string journal_title_;
+    std::string ppn_;
+    std::string online_issn_;
+public:
+    JournalTitlePPNAndOnlineISSN(const std::string &journal_title, const std::string &ppn, const std::string &online_issn)
+        : journal_title_(journal_title), ppn_(ppn), online_issn_(online_issn) { }
+};
+
+
+// Parses an input file that has three (the last component may be empty) parts per line that are colon-separated.
+// Embedded colons may be backslash escaped.
+void LoadISSNsToJournalTitlesPPNsAndISSNsMap(
+    std::unordered_map<std::string, JournalTitlePPNAndOnlineISSN> * const issns_to_journal_titles_ppns_and_issns_map)
+{
+    const std::string MAP_FILE_PATH(UBTools::GetTuelibPath() + "print_issns_titles_online_ppns_and_online_issns.csv");
```

Comment:
If we extend the list or change the row order later we must rename it, maybe also write a migration script and so on... And while reading this file name, when i reach the last part i already forgot what the first part was. Why not simply call the file journals.csv?

Reply:
And why is the column print issns not directly next to the column online issns?